### PR TITLE
fallback nft meta is null

### LINKED ISSUE #2747: Seems newest list from landing is gone
```log
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'id')
    at 41d068c.js:1:10873
    at Array.map (<anonymous>)
    at f.<anonymous> (41d068c.js:1:10838)
    at v (152af89.js:2:16194)
    at Generator._invoke (152af89.js:2:15982)
    at Generator.next (152af89.js:2:16623)
    at r (152af89.js:2:2747)
    at l (152af89.js:2:2950)
    at 152af89.js:2:3009
    at new Promise (<anonymous>)
```

image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2747
- [ ] <brief_description_of_what_I've_added>

the error is caused by the incrooect nft meta, which id is null.
image

the nft: /rmrk/gallery/10869444-c6017764e7a1d03b5e-MOOSENFT-MAGGIE_MOOSE_29-0000000000000029

expected nft meta is an object with id and image.
image

Would be cool if we can filter the events in  squid.

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by vikiival — CHANGES_REQUESTED** [review]
> fallback

fallback is when you ask one girl for a date but in case you ask another girl just in case the first one would say no

The implementation filters out nfts which does not have meta (which is super strange :)) But the correct fallback is to take `metadata` field and fetch it on the client

**Comment by Jarsen136** [comment]
I mock the empty meta and then it displays all the images and works well without any error.

**Review by vikiival — CHANGES_REQUESTED** [review]
Extend `queries/rmrk/subsquid/lastNftListByEvent.graphql`

1. extend query with metadata

```graphql
query lastNftListByEvent($limit: Int!, $event: Interaction!, $and: [EventWhereInput!]) {
  events(
    limit: $limit
    where: { interaction_eq: $event, AND: $and, nft: {burned_eq: false} }
    orderBy: timestamp_DESC
  ) {
    meta
    timestamp
    nft {
      id
      name
      issuer
      currentOwner
      metadata
      meta {
        id
        image
      }
    }
  }
}

```
2. fetch metadata
3. update cache
4. fetch image via cloudflare image

**Comment by yangwao** [comment]
lol probably accidentally found that moment lol and seems working

image

image

**Comment by Jarsen136** [comment]
> lol probably accidentally found that moment lol and seems working
> 
> image
> 
> image

 yup, you caught the same error again 😄

**Comment by Jarsen136** [comment]
> Extend `queries/rmrk/subsquid/lastNftListByEvent.graphql`
> 
> 1. extend query with metadata
> 
> ```graphql
> query lastNftListByEvent($limit: Int!, $event: Interaction!, $and: [EventWhereInput!]) {
>   events(
>     limit: $limit
>     where: { interaction_eq: $event, AND: $and, nft: {burned_eq: false} }
>     orderBy: timestamp_DESC
>   ) {
>     meta
>     timestamp
>     nft {
>       id
>       name
>       issuer
>       currentOwner
>       metadata
>       meta {
>         id
>         image
>       }
>     }
>   }
> }
> ```
> 
> 2. fetch metadata
> 3. update cache
> 4. fetch image via cloudflare image

Now it works for me but I'm not sure if the code meets your expectation. 
The image url has been fetched appropriately.
And I still don't know should we try to fetch `meta.image` if 'meta' is null. 😕

File: components/rmrk/Gallery/LatestSales.vue [hunk 1]
```diff
@@ -74,6 +75,17 @@ export default class LatestSales extends Vue {
   protected async handleResult({ data }: any) {
     this.events = data.events
     this.total = data.events.length
+
```
Comment (vikiival) [inline components/rmrk/Gallery/LatestSales.vue:78]: avoid mutating data

File: components/rmrk/Gallery/NewestList.vue [hunk 2]
```diff
@@ -78,9 +79,19 @@ export default class NewestList extends Vue {
   }
 
   protected async handleResult({ data }: any) {
+    for (const event of data.events) {
```
Comment (vikiival) [inline components/rmrk/Gallery/NewestList.vue:82]: DRY

**Review by vikiival — CHANGES_REQUESTED** [review]

File: components/rmrk/Gallery/LatestSales.vue (hunk 1 above)
Comment (Jarsen136) [reply components/rmrk/Gallery/LatestSales.vue:78]: okay

**Review by vikiival — COMMENTED** [review]
👀

**Review by vikiival — APPROVED** [review]
YOLO MODE ON.

**Comment by vikiival** [comment]
pay 99 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $99 @ 156.21 USD/KSM ~ 0.634 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x9584cedba95dc54eb2dec18c4c0e176c21feb8182f809f9a566acc51f0d67bb2](https://kusama.subscan.io/extrinsic/0x9584cedba95dc54eb2dec18c4c0e176c21feb8182f809f9a566acc51f0d67bb2)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**